sdk/communication/Azure.Communication.Common/src/ClientDiagnosticsExtension.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Communication
+{
+    internal static class ClientDiagnosticsExtension
+    {
+        internal static T RunScoped<T>(this ClientDiagnostics clientDiagnostics, Func<T> func, string className, string methodName)
+        {
+            using DiagnosticScope scope = clientDiagnostics.CreateScope($"{className}.{methodName}");
```

Comment:
@christothes I vaguely remember we had a chat about this. How do you feel about moving this to Azure.Core. Also @pakrym

Reply:
I don't recall the chat, but it looks interesting. The only concern I'd have is that I believe the lambda causes an additional allocation because of the captured variables in the body.

Reply:
Yes, the point of having `clientDiagnostics.CreateScope` in every specific method is to avoid both allocations of lambda and the scope string when diagnostics are not enabled.

Reply:
That is a really good point. I did not think about the allocations. I cannot believe I am saying this, but I kinda wish we could have the capture closure for C++ lambda and make the lambda's static. @pavelprystinka I agree with the folks. Let's not do this change

Reply:
We are also open to ideas on how we can make the code less verbose without losing the perf benefit. It's has been on my mind for a while now.

Reply:
@RezaJooyandeh Maybe one day C# will have static lambdas. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/static-anonymous-functions

Reply:
They are in C# 9 https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9